### PR TITLE
[Network] az network vpn-gateway create: add validation to check the existence of an vpn gateway.

### PR DESCRIPTION
**Pull request by MyronFanQiu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

Fix Azure/azure-cli#1636

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/virtual-wan/azext_vwan/custom.py [hunk 1]
```diff
@@ -390,7 +390,14 @@ def _v3_route_table_client(cli_ctx):
 def create_vpn_gateway(cmd, resource_group_name, gateway_name, virtual_hub,
                        location=None, tags=None, scale_unit=None,
                        asn=None, bgp_peering_address=None, peer_weight=None, no_wait=False):
+    from msrestazure.azure_exceptions import CloudError
+    from .vendored_sdks.v2018_08_01.v2018_08_01.models.error_py3 import ErrorException
     client = network_client_factory(cmd.cli_ctx).vpn_gateways
+    try:
+        client.get(resource_group_name, gateway_name)
+        raise CLIError('{} VPN gateway already exist. Please delete it first.'.format(gateway_name))
```
Comment (haroldrandom) [inline src/virtual-wan/azext_vwan/custom.py:398]: raise anyway?

**Comment by yonzhan** [comment]
add to S172

File: src/virtual-wan/azext_vwan/custom.py (hunk 1 above)
Comment (MyronFanQiu) [reply src/virtual-wan/azext_vwan/custom.py:398]: If resource doesn't exist, `client.get` will raise other exception and go through the `except` logic.

Comment (haroldrandom) [reply src/virtual-wan/azext_vwan/custom.py:398]: How about:

```Python
try:
    client.get(resource_group_name, gateway_name)
    # raise CLIError('{} VPN gateway already exist. Please delete it first.'.format(gateway_name))
except (CloudError, ErrorException):
    pass
else:
    raise CLIError('{} VPN gateway already exist. Please delete it first.'.format(gateway_name))
```

Comment (MyronFanQiu) [reply src/virtual-wan/azext_vwan/custom.py:398]: will do this change

**Pull request merged**